controllers/machine_controller.go

```diff
@@ -657,12 +657,32 @@ func (r *MachineReconciler) nodeToMachine(o client.Object) []reconcile.Request {
 		panic(fmt.Sprintf("Expected a Node but got a %T", o))
 	}
 
+	// Match by nodeName and status.nodeRef.name.
+	filters := []client.ListOption{
+		client.MatchingFields{clusterv1.MachineNodeNameIndex: node.Name},
+	}
+
+	// Match by clusterName when the node has the annotation.
+	clusterName, ok := node.GetAnnotations()[clusterv1.ClusterNameAnnotation]
+	if ok {
+		filters = append(filters,
+			client.MatchingLabels{
+				clusterv1.ClusterLabelName: clusterName,
+			},
+		)
+	}
```

Comment:
Add a test?

Reply:
These annotations were added later, so I'd like to make sure we cover:

- A node with none of these annotations
- A node with a single annotation
- A node with both annotations